### PR TITLE
Fail for Database.update! when git is not available

### LINKED ISSUE #158: Database.update! shall fail when git not installed
I am using the tool (thanks for providing it) within an alpine linux docker container.

I wondered why the update feature did not work. 
It failed without error.

The reason is, that `Database.update!` fails silently when git is not installed (https://github.com/rubysec/bundler-audit/blob/master/lib/bundler/audit/database.rb#L94-L104)
- `.update!` should check if `git` is available
- `.update!` should return false or an error when `git` is not available on the system

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #158

### CODE REVIEW

**Comment by pxlpnk** [comment]
Raising an exception and failing with a stack trace might be not a good way of handling this situation.
A more and better defined way would be to print an error message and end the process with an exist status `!=0`

**Comment by fatkodima** [comment]
@pxlpnk Updated.
Now `cli.rb` checks that git is not installed.

Or should `Database.update!` still raise some custom error and in `cli.rb` we rescue it and print message?

**Comment by pxlpnk** [comment]
I think its good the way is it now. Maybe in the future with different output formaters this would have to be changed.

**Comment by pxlpnk** [comment]
@postmodern is there a way to get this merged?

**Comment by and0x000** [comment]
possibly related to https://github.com/rubysec/bundler-audit/issues/175 where it is also mentioned, that updating the DB is skipped.

**Comment by pxlpnk** [comment]
@reedloden How do we get contributions into bundler audit?

**Comment by ZimbiX** [comment]
LGTM; I'd probably have used an exception as mentioned (whenever a call to Git is made where it's not installed), but this looks fine to me. Definitely good not to be giving people a false sense of security. Cheers for the fix

@postmodern Merging would be much appreciated =) Thanks

**Comment by reedloden** [comment]
Can we split the other changes and the DB update?

**Comment by fatkodima** [comment]
@reedloden Updated.

File: spec/bundle/insecure_sources/Gemfile [hunk 1]
```diff
@@ -1,4 +1,4 @@
 source 'http://rubygems.org'
 
-gem 'rails'
+gem 'rails', '~> 5.2'
```
Comment (reedloden) [inline spec/bundle/insecure_sources/Gemfile:3]: This doesn’t seem related. Can this be removed?

File: spec/cli_spec.rb [hunk 2]
```diff
@@ -46,6 +46,33 @@
         end
 
       end
+
+      context "when git is not installed" do
+        before do
+          expect(Bundler::Audit::Database).to receive(:update!).and_return(nil)
```
Comment (reedloden) [inline spec/cli_spec.rb:52]: The specs were all updated recently. Please see about conforming this and any other code to the recent style changes. For example, probably can use described_class here?

File: spec/bundle/insecure_sources/Gemfile (hunk 1 above)
Comment (fatkodima) [reply spec/bundle/insecure_sources/Gemfile]: removed

File: spec/cli_spec.rb (hunk 2 above)
Comment (fatkodima) [reply spec/cli_spec.rb:52]: `described_class` is `Bundler::Audit::CLI` in this case, so cannot be used
I do not see inconsistencies with surrounding code.

Comment (reedloden) [reply spec/cli_spec.rb:52]: Thanks! I was reviewing from my phone, so I couldn't see full file context. :-)

**Pull request merged**